src/operators/courseItemOperations.ts

```diff
@@ -0,0 +1,61 @@
+import CourseItem, { CourseItemInterface } from '../models/courseItem';
+import { v4 as uuidv4 } from 'uuid';
+
+export const create = async (courseItem: CourseItemInterface) => {
+	try {
+		const hashKey = uuidv4();
+		courseItem.id = hashKey;
+		console.log('id: ' + courseItem.id);
+		await CourseItem.create(courseItem);
+	} catch (err) {
+		console.error(err);
+	}
+};
+
+// NOTE: any reference to key is referring to the hash key, or the `id` parameter in all the models.
+export const read = async (key: string, user: string) => {
+	try {
+		const courseItem = await CourseItem.get(key);
+		if (courseItem && courseItem.owner == user) {
+			return courseItem;
+		} else {
+			throw 'ERROR - could not read courseItem with key ' + key;
+		}
+	} catch (err) {
+		console.error(err);
+	}
+};
+
+export const update = async (
+	data: Partial<CourseItemInterface>,
+	userID: string
+) => {
+	try {
+		if (data.id) {
+			const key = data.id;
+			if (await checkUserID(key, userID)) {
+				await CourseItem.update(data);
+			} else {
+				throw "ERROR - userID doesn't match";
+			}
+		}
+	} catch (err) {
+		console.error(err);
+	}
+};
+
+export const del = async (key: string, userID: string) => {
+	try {
+		if (checkUserID(key, userID)) {
+			await CourseItem.delete(key);
+			console.log('Deletion of document with id ' + key + ' successful.');
+		}
+	} catch (err) {
+		console.error(err);
+	}
+};
+
+const checkUserID = async (key: string, userID: string) => {
+	const courseItem = await CourseItem.get(key);
+	return courseItem.owner == userID;
+};
```

Comment:
I like this. We should make it generic for everything:

```ts
const checkUserId = 
(getDocument: (key:string) => DocumentType) => 
async (key: string, userID: string) => {
    const document = await getDocument(key)
    if(document.owner) {
        return document.owner === userID;
    }
    return document.id === userID;
}
```
where DocumentType is the union ( A | B) type of all the possible documents that we support.

Then for anyone to use it they could do this:

```ts
import { checkUserId } from './index.ts' // Maybe a good spot?

const checkSemesterUser = checkUsedId(Semester.get);

await checkSemesterUser(key, userId);

```

The only reason I say this is because it removes the duplicated logic from everyones files.

If what is happening doesn't make sense to you, 1: Read [this article](https://medium.com/javascript-scene/curry-and-function-composition-2c208d774983) 2. Ask me :)

Reply:
Great suggestion and article - I'm glad I know of this now!
Just a quick question:
Why add the `if` statement? For what purpose would we want to check if `userID === document.id`?
(Re:)
```
if(document.owner) {
    return document.owner === userID;
}
return document.id === userID;
```

Reply:
Also totally open-ended question, do you prefer `userID` or `userId`?